codegenerator/cli/templates/static/codegen/src/eventFetching/FetchState.res

```diff
@@ -561,53 +561,17 @@ let minOfOption: (int, option<int>) => int = (a: int, b: option<int>) => {
   }
 }
 
-/**
-Constructs `nextQuery` from a given node
-*/
-let getNextQueryFromNode = (
-  {registerType, latestFetchedBlock, contractAddressMapping}: register,
-  ~toBlock,
-  ~partitionId,
-) => {
-  let (id, endBlock) = switch registerType {
-  | RootRegister({endBlock}) => (Root, endBlock)
-  | DynamicContractRegister({id}) => (DynamicContract(id), None)
-  }
-  let fromBlock = switch latestFetchedBlock.blockNumber {
+let getNextFromBlock = ({latestFetchedBlock}: register) => {
+  switch latestFetchedBlock.blockNumber {
   | 0 => 0
   | latestFetchedBlockNumber => latestFetchedBlockNumber + 1
   }
-  let toBlock = minOfOption(toBlock, endBlock)
-  {
-    partitionId,
-    fetchStateRegisterId: id,
-    fromBlock,
-    toBlock,
-    contractAddressMapping,
-  }
 }
 
-type nextQueryOrWaitForBlock =
+type nextQueryOrDone =
   | NextQuery(nextQuery)
-  | WaitForNewBlock
```

Comment:
Moved all the interactions with global state to the global state module. Here we simply prepare a query and don't care how it'll be executed.